components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/GatewayPolicyDeployer.java

```diff
@@ -0,0 +1,187 @@
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.context.MessageContext;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.gateway.GatewayPolicyDTO;
+import org.wso2.carbon.apimgt.gateway.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.gateway.service.APIGatewayAdmin;
+import org.wso2.carbon.apimgt.impl.dto.GatewayArtifactSynchronizerProperties;
+import org.wso2.carbon.apimgt.impl.gatewayartifactsynchronizer.ArtifactRetriever;
+import org.wso2.carbon.apimgt.impl.gatewayartifactsynchronizer.exception.ArtifactSynchronizerException;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * This class contains the methods used to retrieve artifacts from a storage and deploy and undeploy
+ * the gateway policy mappings in gateway.
+ */
+public class GatewayPolicyDeployer {
+    private static final Log log = LogFactory.getLog(GatewayPolicyDeployer.class);
+    ArtifactRetriever artifactRetriever;
+    private String gatewayPolicyMappingUuid;
+    GatewayArtifactSynchronizerProperties gatewayArtifactSynchronizerProperties;
+
+    public GatewayPolicyDeployer(String gatewayPolicyMappingUuid) {
+        this.artifactRetriever = ServiceReferenceHolder.getInstance().getArtifactRetriever();
+        this.gatewayPolicyMappingUuid = gatewayPolicyMappingUuid;
+        this.gatewayArtifactSynchronizerProperties = ServiceReferenceHolder
+                .getInstance().getAPIManagerConfiguration().getGatewayArtifactSynchronizerProperties();
+    }
+
+    public GatewayPolicyDeployer() {
+        this.artifactRetriever = ServiceReferenceHolder.getInstance().getArtifactRetriever();
+        this.gatewayArtifactSynchronizerProperties = ServiceReferenceHolder
+                .getInstance().getAPIManagerConfiguration().getGatewayArtifactSynchronizerProperties();
+    }
+
+    /**
+     * Deploy gateway policy in the gateway using the deployGatewayPolicy method in gateway admin.
+     *
+     * @throws ArtifactSynchronizerException
+     */
+    public void deployGatewayPolicyMapping() throws ArtifactSynchronizerException {
+        try {
+            GatewayPolicyDTO gatewayPolicyDTO = retrieveGatewayPolicyArtifact(gatewayPolicyMappingUuid);
+            if (gatewayPolicyDTO != null) {
+                APIGatewayAdmin apiGatewayAdmin = new APIGatewayAdmin();
+                MessageContext.setCurrentMessageContext(
+                        org.wso2.carbon.apimgt.gateway.utils.GatewayUtils.createAxis2MessageContext());
+                apiGatewayAdmin.deployGatewayPolicy(gatewayPolicyDTO);
+            } else {
+                log.error("Gateway policy artifact is not found for the policy mapping: " + gatewayPolicyMappingUuid);
+            }
+        } catch (AxisFault e) {
+            throw new RuntimeException(e);
```

Comment:
Any reason for throwing runtime exception?

Reply:
resolved with https://github.com/wso2/carbon-apimgt/pull/12201/commits/4042971246cf43e60948916f7de0a263bb8f2344

thank you